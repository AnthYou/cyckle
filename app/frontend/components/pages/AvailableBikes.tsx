import React, { useEffect, useState } from "react";
import { Bike } from "@/utils/interfaces";
import useHttp from "@/hooks/use-http";
import BikeCard from "../bike/Bike";
import LoadingSpinner from "../UI/LoadingSpinner";
import { Link } from "react-router-dom";

const AvailableBikes = () => {
  const [bikes, setBikes] = useState<Bike[]>([]);
  const { isLoading, error, sendRequest: fetchBikes } = useHttp();

  useEffect(() => {
    fetchBikes(
      {
        url: "/api/v1/bikes",
      },
      (responseJson) => {
        const fetchedBikes: Bike[] = responseJson.data;
        setBikes(fetchedBikes);
      }
    );
  }, [fetchBikes]);

  return (
    <>
      <h1 className="mb-7">Quel v√©lo allez-vous louer aujourd'hui ?</h1>
      {isLoading && <LoadingSpinner />}
      {error && <p>{error}</p>}
      <div className="grid sm:grid-cols-2 md:grid-cols-3 gap-4">
        {bikes.map((bike) => (
          <Link key={bike.id} to={`/bikes/${bike.id}`}>
            <BikeCard {...bike} />
          </Link>
        ))}
      </div>
    </>
  );
};

export default AvailableBikes;
