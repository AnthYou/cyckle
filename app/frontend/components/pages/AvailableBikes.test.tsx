import React from "react";
import renderer from "react-test-renderer";

import AvailableBikes from "./AvailableBikes";

describe('<AvailableBikes />', () => {
  it('should render correctly', () => {
    const component = renderer.create(<AvailableBikes />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  })
});
