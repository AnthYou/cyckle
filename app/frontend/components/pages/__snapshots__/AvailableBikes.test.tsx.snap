// Vitest Snapshot v1

exports[`<AvailableBikes /> > should render correctly 1`] = `
<div
  className="grid sm:grid-cols-2 md:grid-cols-3 gap-4"
/>
`;
